File: bms/go/geo/geo_s1_ChordAngle_Tan/geo_s1_ChordAngle_Tan.go
package test

import (
	"math"
)

func (c geo_s1_ChordAngle) Tan() float64 {
	return c.Sin() / c.Cos()
}

func (c geo_s1_ChordAngle) Sin() float64 {
	return math.Sqrt(c.Sin2())
}

func (c geo_s1_ChordAngle) Sin2() float64 {
	// Let a be the (non-squared) chord length, and let A be the corresponding
	// half-angle (a = 2*sin(A)). The formula below can be derived from:
	//   sin(2*A) = 2 * sin(A) * cos(A)
	//   cos^2(A) = 1 - sin^2(A)
	// This is much faster than converting to an angle and computing its sine.
	return float64(c * (1 - 0.25*c))
}

func (c geo_s1_ChordAngle) Cos() float64 {
	// cos(2*A) = cos^2(A) - sin^2(A) = 1 - 2*sin^2(A)
	return float64(1 - 0.5*c)
}

type geo_s1_ChordAngle float64

File: translations_only/fail/go/geo/geo_s1_ChordAngle_Tan/claude3/Restart/geo_s1_ChordAngle_Tan.rs

use std::f64::consts::PI;

fn tan(c: f64) -> f64 {
    sin(c) / cos(c)
}

fn sin(c: f64) -> f64 {
    (sin2(c)).sqrt()
}

fn sin2(c: f64) -> f64 {
    // Let a be the (non-squared) chord length, and let A be the corresponding
    // half-angle (a = 2*sin(A)). The formula below can be derived from:
    //   sin(2*A) = 2 * sin(A) * cos(A)
    //   cos^2(A) = 1 - sin^2(A)
    // This is much faster than converting to an angle and computing its sine.
    c * (1.0 - 0.25 * c)
}

fn cos(c: f64) -> f64 {
    // cos(2*A) = cos^2(A) - sin^2(A) = 1 - 2*sin^2(A)
    1.0 - 0.5 * c
}
